datafusion/physical-expr/src/window/built_in.rs

```diff
@@ -157,18 +162,20 @@ impl WindowExpr for BuiltInWindowExpr {
             };
             let state = &mut window_state.state;
 
-            let (values, order_bys) =
-                self.get_values_orderbys(&partition_batch_state.record_batch)?;
+            let batch_ref = &partition_batch_state.record_batch;
+            let mut values = self.evaluate_args(batch_ref)?;
+            let order_bys = if evaluator.uses_window_frame() | evaluator.include_rank() {
```

Comment:
I wonder if this should be the `||` boolean operator rather than `|` bitwise (though perhaps it doesn't matter as they  are probably the same for booleans)

Reply:
Changed it